catalog/src/main/java/org/polypheny/db/catalog/CatalogImpl.java

```diff
@@ -405,6 +431,68 @@ public void restoreColumnPlacements( Transaction transaction ) {
     }
 
 
+    /**
+     * on restart all RelNodes used in views and materialized views need to be recreated
+     * depending on the query language different methods are used
+     */
+    @Override
+    public void restoreViews( Transaction transaction ) {
+        Statement statement = transaction.createStatement();
+
+        for ( CatalogTable c : tables.values() ) {
+            if ( c.tableType == TableType.VIEW || c.tableType == TableType.MATERIALIZEDVIEW ) {
+                String query;
+                QueryLanguage language;
+                if ( c.tableType == TableType.VIEW ) {
+                    query = ((CatalogView) c).getQuery();
+                    language = ((CatalogView) c).getLanguage();
+                } else {
+                    query = ((CatalogMaterialized) c).getQuery();
+                    language = ((CatalogMaterialized) c).getLanguage();
+                }
+
+                switch ( language ) {
+                    case SQL:
+                        SqlProcessor sqlProcessor = statement.getTransaction().getSqlProcessor();
+                        SqlNode sqlNode = sqlProcessor.parse( query );
+                        RelRoot relRoot = sqlProcessor.translate(
+                                statement,
+                                sqlProcessor.validate( statement.getTransaction(), sqlNode, RuntimeConfig.ADD_DEFAULT_VALUES_IN_INSERTS.getBoolean() ).left );
+                        nodeInfo.put( c.id, relRoot.rel );
+                        relTypeInfo.put( c.id, relRoot.validatedRowType );
+                        break;
+                    case RELALG:
+                        Gson gson = new Gson();
+
+                        RelNode result = QueryPlanBuilder.buildFromTree( gson.fromJson( query, UIRelNode.class ), statement );
```

Comment:
Refactor

Reply:
Added aditional interface JsonRelProcessor in order to translate the query into a RelNode.